test/langswitch/LanguageSwitchControllerSpec.scala

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2025 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package langswitch
+
+import langswitch.LanguageSwitchController
+import langswitch.Languages.English
+import play.api.http.Status
+import play.api.test.FakeRequest
+import play.api.test.Helpers.{defaultAwaitTimeout, status}
+import support.AppSpec
+
+final class LanguageSwitchControllerSpec extends AppSpec {
+  private val controller = app.injector.instanceOf[LanguageSwitchController]
+
+  "switch to language should return SEE OTHER" in {
+    val fakeRequest = FakeRequest("GET", "/")
+
+    val result = controller.switchToLanguage(English)(fakeRequest)
+
+    status(result) shouldBe Status.SEE_OTHER
```

Comment:
it'd be nice to test what the redirect location is and the language in the result as well. There seem to be two cases which result in different behaviour in this controller endpoint - one where there's a referrer header and one where there's not. As a general rule, it'd be good to test all the different results you can get with whatever you're testing.